### PR TITLE
[GH-435] Supercampaigns scene

### LINKED ISSUE #435: [FEATURE] [DUNGEON] Supercampaigns scene
### Motivation

The current implementation of the CampaignsMap scene is quite hardcoded, it should be adapted to represent any supercampaign it receives. 

### Implementation

- Dynamically create the campaigns icons.
- Change the scene's background according the supercampaign that is being played.
- Make the necessary changes in the Overworld to send the main supercampaign to this scene only.
- Back button should take back to the correct scene (see #434)

### PR DESCRIPTION
**Pull request by ncontinanza**
Closes #435 

## Motivation

We want to be able to generalize the visualization of supercampaigns, in order to avoid having different scenes for each supercampaign.

## Summary of changes

- Implement supercampaigns scene as an almost empty scene, consisting only of a SceneManager and a SuperCampaignsMapManager.
- Implement the content of the supercampaigns scene as prefabs. Currently we only have the main campaign's prefab, but soon a prefab for the dungeon will be implemented.

## How has this been tested?

Start the game and go to the campaigns from the overworld, you should see no changes and everything should still be working as usual.
You can also create a new supercampaign following the documentation found in `docs/how_to_add_supercampaigns.md`

## Checklist
- [X] I have tested the changes locally.
- [X] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by tkz00 — CHANGES_REQUESTED** [review]
Screenshot 2024-04-08 at 4 32 47 PM

The back button isn't visible in the screen, I couldn't take a full look at why, but I know it works fine in the campaign scene, tomorrow we can check that out and fix this scene.

File: client/Assets/Scripts/CampaignsMap/SupercampaignsMapManager.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    [SerializeField]
+    LevelManager sceneManager;
+    private GameObject supercampaignPrefab;
+    public static string selectedSuperCampaignName;
+
+    void Start()
+    {
+        GameObject selectedPrefab = Resources.Load("Supercampaigns/" + selectedSuperCampaignName) as GameObject;
+        supercampaignPrefab = Instantiate(selectedPrefab, new Vector3(0, 0, 0), Quaternion.identity);
+
```
Comment (tkz00) [inline client/Assets/Scripts/CampaignsMap/SupercampaignsMapManager.cs:15]: I think there are better alternatives to using `Resources.Load`, I haven't used them yet, but I think Addressables are a better alternative for what we are trying to achieve here, even if this isn't the answer I would look for other options.

**Review by tkz00 — COMMENTED** [review]
I think objects that don't vary with the supercampaign, such as the canvas, the header or the back button, shouldn't be inside the supercampaign prefab, mainly for maintenance and organizational reasons.

image

**Review by tkz00 — CHANGES_REQUESTED** [review]
While the Main supercampaign prefab works, the template has some transform properties that need to be changed: the rect transform of the parent object, the Map object and the Campaigns object all should have the same values as the Main prefab.

This causes the new prefab variants for supercampaigns to not display correctly when they should.

**Review by tkz00 — CHANGES_REQUESTED** [review]
The `Supercampaign` scriptable object is unused and should be removed. Also the ElMessiri font shouldn't be added in this PR.

**Review by tkz00 — CHANGES_REQUESTED** [review]
Screenshot 2024-05-23 at 5 34 19 PM

The images for the victory rewards seem to be broken.

**Review by tkz00 — APPROVED** [review]
Works great!

**Review by lotuuu — APPROVED** [review]

**Review by tkz00 — APPROVED** [review]
I think it would be interesting to be able to dynamically instantiate the required campaign without having to add an if statemenet on `SupercampaignsMapManager.cs` and a method on `AddressableInstantiator.cs` for each supercampaign, but for now I feel its good, and maybe what I said won't ever be necessary.

**Pull request merged**